feat(core): adding vibe coding blog (#61)
ZephyrCloudIO/zephyr-website: src/routes/blog/[id$]/vibe-coding.mdx

By combining intelligent agents with cutting-edge technologies like Module Federation and Micro Frontends, they’ve redefined how software is developed and deployed.
>
> This is more than just automation — it’s a modular, scalable, and intelligent approach that enables end-to-end product delivery with unprecedented speed and efficiency.
It’s a whole new way of thinking about development.
>
> The future of software is autonomous, distributed, and already in motion — and Zephyr is leading the way.
---
#### Alex Bennett: Accuris
> Zephyr Cloud makes the dev loop so smooth that your favorite AI tools can jump right in–no set up needed. It’s not just about helping people move faster, it’s about gearing up the next-gen AI workforce too.
---
#### Glauber Costa: Turso
> sub-second deployments are freaking awesome! pairs well with sub-second database deployments

## Overview
### What we built?
When thinking of what to build I wanted something that was instantly recognizable but also had a little bit of complexity.
I decided to work on a quick clone of X. This could be something that could be worked on intermittently throughout the week before launch week.

> The clone is consists of 6 micro frontends, each responsible for a distinct feature or UI segment

Reviewer: ```suggestion
The clone consists of 6 micro frontends, each responsible for a distinct feature or UI segment:
```

* App Shell
* Feed
* Grok
* Create
* Verified Organizations
* Explore

### How did I get there
Claude code is one of my favorite tools when it comes to vibe coding, call me old-fashioned but nothing beats opening up IntelliJ and typing `claude code` in the terminal.

> After creating a directory quick I had claude go and generate a Nx workspace with some baseline tools I like to use like commitlint and husky.

Reviewer: ```suggestion
After quickly creating a directory I had Claude go and generate an Nx workspace with some baseline tools I like to use like commitlint and husky.
```

however with now over 3m downloads a week Module Federation it is safe to say it is now the conical choice.

What other benefits do we get from MF + MFEs?
* Independent deployments and modular code ownership.
* Scalability and faster iteration by splitting the app into smaller, separately deployable chunks.
* Smaller context window for each of the AI Agents

## Our Artificial Friends aka AI Tools & How They Fit In
### Trae
One of the new kids on the block [Trae](https://www.trae.ai/) (The Real AI Engineer) is an AI based IDE created by the ByteDance team.
Since I have friends working on this project I choose to play with it on this project to see how it would do on the MFEs with simple tasks.
While under the covers it was still using Claude Sonnet 3.7. Trae gave me really easy to use changes and was able to understand the repo really well.

### Anthropic Claude Code
We are lucky enough to have some credits from the startup program to use [Claude Code](https://docs.anthropic.com/en/docs/agents-and-tools/claude-code/overview).
This is the current tool of choice for most of our team at Zephyr when it comes to agentic coding.

### Devin

> A cloud based AI development tool [Devin](https://devin.ai/) made waves when they first started.

Reviewer: ```suggestion
A cloud-based AI development tool [Devin](https://devin.ai/) made waves when they first started.
```

I also had the opportunity to meet their team at the True Ventures summit and one of the issues they are seeing in the community is people understanding the impact of Devin's changes.
With Zephyr we hope to fix that by giving you realtime visibility into its work.

> There was two rather intense moments when using Devin:

Reviewer: ```suggestion
There were two rather intense moments when using Devin:
```

1) When using Devin to code remotely I was able to see its changes intermingled into my view of production without actually deploying it. See our promo video about this one above.
2) Devin actually understood the CLI output and its value and gave it to me as the end user.

<img src={Devin} alt='Devin understanding the Zephyr output' />

## Key Takeaways
### Faster Iteration With AI
* When you have multiple AI tools assisting with code scaffolding, debugging, and testing, you really can have higher output.

> * The synergy between developer creativity and AI suggestions speeds up the typical dev cycle especially when you can see their work in realtime.

Reviewer: ```suggestion
* The synergy between developer creativity and AI suggestions speeds up the typical dev cycle especially when you can see their work in real-time.
```
